### ISSUE TITLE
Training image loading problems

### ISSUE THREAD

**Issue by Chenhanne** [open-mmlab/mmdetection#11196]
I use a custom coco dataset for training and there are instances where the training images are not fully loaded.
I have 2583 training pictures and 884 test pictures.
As shown, only half of the training images can be detected. But all the test images can be detected.

![image](https://github.com/open-mmlab/mmdetection/assets/116365899/43542dae-515b-4a5b-aa0d-a633df09752d)

To find the problem, I used the training set as a test set and the test set as a training set to check if it was a problem with the data that caused the images to load incompletely. However, the results are as follows:

![image](https://github.com/open-mmlab/mmdetection/assets/116365899/996eacf9-c51c-438f-bb46-0b1d0db71f42)
![image](https://github.com/open-mmlab/mmdetection/assets/116365899/805f28f8-518f-4b58-b5ca-fa063e0693e2)

It is easy to see that the original training set was able to be fully loaded (2583 images) after being used as a test set. However, the original test set was only half recognised (442 images) after being used as a training set.
It seems that only the training process has the problem
What's going on here? How can I solve this problem? Thank you very much.

Besides, the model I use is:
base = [
'../base/models/faster-rcnn_r50_fpn.py',
'../base/datasets/coco_detection.py',
'../base/schedules/schedule_1x.py',
'../base/default_runtime.py'
]

**Comment by GeorgePearse** [comment]
Those numbers are actually the numbers of batches that the model needs to go through to perform an epoch's worth of training, any chance you have a training batch size of 1, and test batch size of 2?

**Comment by Chenhanne** [comment]
Thank you very much for your answer! I understand! You are right!👍👍👍

---- Replied Message ----
| From | George ***@***.***> |
| Date | 11/21/2023 19:19 |
| To | open-mmlab/mmdetection ***@***.***> |
| Cc | Chenhanne ***@***.***>,
Author ***@***.***> |
| Subject | Re: [open-mmlab/mmdetection] Training image loading problems (Issue #11196) |

Those numbers are actually the numbers of batches that the model needs to go through to perform an epoch's worth of training, any chance you have a training batch size of 1, and test batch size of 2?
